### PR TITLE
Adding GHA workflow to sync with Jira

### PR DESCRIPTION
**Pull request by mmcquillan**
This is a GHA workflow to allow labeling individual GitHub issues to synchronize into Jira.

### CODE REVIEW

File: .github/workflows/jira-sync.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+          JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+
+      - name: Search
+        if: github.event.action != 'opened'
+        id: search
+        uses: tomhjp/gh-action-jira-search@04700b457f317c3e341ce90da5a3ff4ce058f2fa # v0.2.2
+        with:
+          # cf[10089] is Issue Link (use JIRA API to retrieve)
+          jql: 'issuetype = "${{ steps.set-ticket-type.outputs.TYPE }}" and cf[10089] = "${{ github.event.issue.html_url || github.event.pull_request.html_url }}"'
```
Comment (schmichael) [inline .github/workflows/jira-sync.yml:55]: What's this do?

File: .github/workflows/jira-sync.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        uses: atlassian/gajira-login@45fd029b9f1d6d8926c6f04175aa80c0e42c9026 # v3.0.1
+        env:
+          JIRA_BASE_URL: ${{ secrets.JIRA_BASE_URL }}
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+          JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+
+      - name: Set ticket type
+        id: set-ticket-type
+        run: |
+          echo "TYPE=GH Issue" >> $GITHUB_OUTPUT
```
Comment (tgross) [inline .github/workflows/jira-sync.yml:28]: This looks to be hitting a shellcheck lint: https://github.com/hashicorp/nomad/actions/runs/9453086274/job/26037680253?pr=23286 I think we need to quote this?
```suggestion
        run: |
          echo "TYPE=GH Issue" >> "$GITHUB_OUTPUT"
```

Comment (mmcquillan) [reply .github/workflows/jira-sync.yml:28]: Thanks!

**Review by tgross — APPROVED** [review]
LGTM!

**Pull request merged**